### ISSUE TITLE
Can't delete any file from vanilla

### ISSUE THREAD

**Issue by granvaud** [vanilla-music/vanilla#1134]
- Vanilla Music Version (Settings -> About): 1.1.0
- Android Version: 12 (lineageOS)
- Phone vendor / model: SM-G970F
- Application was downloaded from (use `[x]`):
  - [ ] Play Store
  - [x ] F-Droid
  - [ ] Other (Nightly Build)
- Did this work in a previous version?
  - [ ] Yes
  - [ ] No
  - [ x] NA / Don't know.

## Description
I have a failure message when I try to delete a file and I don't know what to do...
....

## Steps to reproduce the issue

...

## Screenshot / Video (If applicable)

...

**Comment by luuuciano** [comment]
Maybe I have this issue too? I am using lineageOS 18.1
I have applied the file permission access... so files are listed, I can listen, etc...
BUT when I try to delete a file it just shows "0 tracks deleted"...

Any idea? What to do? What to check?

Deleting files I do not like was a very used feature here... D-:

**Comment by andiandi13** [comment]
I think it was discussed here #691
Files from internal storage can be deleted, by not from SD Card since Vanilla doesn't have permissions

**Comment by luuuciano** [comment]
Mmmm maybe, but I do not think so... I have no SD card, is the regular internal storage
And I remember Vanilla asking for access to it (it give it), there is no error showed when trying to delete a song, it just does nothing

**Comment by Cwpute** [comment]
Maybe we can try to see how other music players implemented that feature/corrected that bug, and see if any of their method could fit into how Vanilla works. There are plenty other players on Fdroid store to choose from:
![Screenshot_20230107-201737](https://user-images.githubusercontent.com/80604512/211166999-9fcd02f9-c933-4dc7-9298-dba2ea6b73e1.png)
I don't personally have the knowledge for it , but maybe someone here has?

**Comment by andiandi13** [comment]
Interesting, so it's an SDCard access issue. I'm in Android 9 so maybe there's an issue with further versions.

**Comment by OxygenCobalt** [comment]
Developer of [Auxio](https://github.com/oxygencobalt/Auxio) here. Since API 30 there has been a bug where using `MediaStore` to modify ***any*** audio in the SD Card (Including deletion) results in a security error. The only way to resolve it is either to obtain the extremely dangerous `MANAGE_EXTERNAL_STORAGE` permission (See: [Material Files](https://github.com/zhanghai/materialfiles)) or downgrade to API 29.

Repeated attempts to report this issue have been closed by google with the reasoning that it's "Working as Intended". This makes no sense until you realize that google hates the filesystem and deliberately crippling an API like this furthers their quest to turn Android into an iOS-like data funnel to some far away server.

This is personally why I do not implement song deletion in my app. You basically cannot implement it in any reasonable capacity on modern android. You're generally better of removing the ability entirely and recommending the use of a file manager instead.

**Comment by luuuciano** [comment]
The regular phone storage is taken as an external SD card too? Because I do not have the files in a SD card...
Currently I am using "Music Player" 0.9.6 and I can delete files using it...

**Comment by OxygenCobalt** [comment]
> The regular phone storage is taken as an external SD card too? Because I do not have the files in a SD card...

No. Most phone storage is considered "internal" and doesn't really suffer from this issue.

**Comment by ShadyHippo** [comment]
Same issue here, no SD card, but I assume this is just Google tomfoolery trying to make file system difficult. Thanks for this wonderful app. Would be great to be able to delete in app though
